src/uucore/src/lib/features/selinux.rs

```diff
@@ -109,6 +123,77 @@ pub fn set_selinux_security_context(path: &Path, context: Option<&String>) -> Re
     }
 }
 
+/// Gets the SELinux security context for the given filesystem path.
+///
+/// Retrieves the security context of the specified filesystem path if SELinux is enabled
+/// on the system.
+///
+/// # Arguments
+///
+/// * `path` - Filesystem path for which to retrieve the SELinux context.
+///
+/// # Returns
+///
+/// * `Ok(String)` - The SELinux context string if successfully retrieved. Returns an empty
+///   string if no context was found.
+/// * `Err(Error)` - An error variant indicating the type of failure:
+///   - `Error::SELinuxNotEnabled` - SELinux is not enabled on the system.
+///   - `Error::FileOpenFailure` - Failed to open the specified file.
+///   - `Error::ContextRetrievalFailure` - Failed to retrieve the security context.
+///   - `Error::ContextConversionFailure` - Failed to convert the security context to a string.
+///
+/// # Examples
+///
+/// ```
+/// use std::path::Path;
+/// use uucore::selinux::{get_selinux_security_context, Error};
+///
+/// // Get the SELinux context for a file
+/// match get_selinux_security_context(Path::new("/path/to/file")) {
+///     Ok(context) => {
+///         if context.is_empty() {
+///             println!("No SELinux context found for the file");
+///         } else {
+///             println!("SELinux context: {}", context);
+///         }
+///     },
+///     Err(Error::SELinuxNotEnabled) => println!("SELinux is not enabled on this system"),
+///     Err(Error::FileOpenFailure) => println!("Failed to open the file"),
+///     Err(Error::ContextRetrievalFailure) => println!("Failed to retrieve the security context"),
+///     Err(Error::ContextConversionFailure) => println!("Failed to convert the security context to a string"),
+/// }
+/// ```
+
+pub fn get_selinux_security_context(path: &Path) -> Result<String, Error> {
+    if selinux::kernel_support() == selinux::KernelSupport::Unsupported {
+        return Err(Error::SELinuxNotEnabled);
+    }
+
+    let f = match std::fs::File::open(path) {
+        Ok(file) => file,
+        Err(_) => return Err(Error::FileOpenFailure),
+    };
```

Comment:
I think you can simplify it to:
```suggestion
    let f = std::fs::File::open(path).map_err(|_| Error::FileOpenFailure)?;
```